packages/lab/src/__tests__/__e2e__/date-picker/DatePicker.cy.tsx

```diff
@@ -0,0 +1,205 @@
+    });
+    it("THEN should error and not format invalid dates on blur", () => {
+      cy.mount(<Default defaultStartDate={testDate} />);
+      cy.findByRole("textbox").click().clear().type("date");
+      cy.findByRole("textbox").blur();
+      cy.findByRole("textbox").should("have.value", "date");
+    });
+    it("THEN clicking the calendar button should open the panel", () => {
+      cy.mount(<Default defaultStartDate={testDate} />);
+      cy.findByRole("button", { name: "Open Calendar" }).realClick();
+      cy.findByRole("application").should("exist");
+    });
+    it("THEN should close the calendar panel once a date is selected", () => {
+      cy.mount(<Default defaultStartDate={testDate} />);
+      cy.findByRole("button", { name: "Open Calendar" }).realClick();
+      cy.findByRole("button", {
+        name: formatDay(testDate.add({ days: 11 })),
+      }).should("exist");
+      cy.findByRole("button", {
+        name: formatDay(testDate.add({ days: 11 })),
+      }).realClick();
+      cy.findByRole("application").should("not.exist");
+    });
+    it("THEN should not open the calendar when disabled", () => {
+      cy.mount(<Default defaultStartDate={testDate} disabled />);
+      cy.findByRole("button").should("be.disabled");
+    });
+    it("THEN it should update the selected month when changing selected date", () => {
+      cy.mount(<Default defaultStartDate={testDate} />);
+      cy.findByRole("textbox").click().clear().type(testInput);
+      cy.findByRole("textbox").blur();
+      cy.findByRole("button", { name: "Open Calendar" }).realClick();
+      cy.findByRole("button", {
+        name: "Next Month",
+      }).focus();
+    });
+    it("THEN it should clear the date if an empty input is submitted", () => {
+      cy.mount(
+        <Default
+          defaultStartDate={testDate}
+          // @ts-ignore
+          CalendarProps={{ visibleMonth: testDate }}
+        />
+      );
+      cy.findByRole("textbox").click().clear();
+      cy.findByRole("textbox").blur();
+      cy.findByRole("button", { name: "Open Calendar" }).realClick();
+      cy.findByRole("button", { name: formatDay(testDate) }).should(
+        "not.have.attr",
+        "aria-pressed",
+        "true"
+      );
+    });
+  });
+  describe("WHEN range datepicker is mounted", () => {
+    it("THEN it should mount with the specified defaultStartDate and defaultEndDate", () => {
+      cy.mount(
+        <Range
+          defaultStartDate={testDate}
+          defaultEndDate={testDate.add({ months: 1 })}
+        />
+      );
+      cy.findAllByRole("textbox")
+        .eq(0)
+        .should("have.value", formatInput(testDate));
+      cy.findAllByRole("textbox")
+        .eq(1)
+        .should("have.value", formatInput(testDate.add({ months: 1 })));
+    });
+
+    it("THEN it should allow selecting start and end date trough input", () => {
+      cy.mount(<Range />);
+      cy.findAllByRole("textbox").eq(0).clear().click().type(testInput);
```

Comment:
This makes me wonder whether label on 2 inputs should be different, indicating one is start and one is end?